Actually print something useful if the JS engine can't be started for some
reason.

--- js/src/shell/js.cpp.orig
+++ js/src/shell/js.cpp
@@ -8185,7 +8185,8 @@
   SetOutputFile("JS_STDERR", &rcStderr, &gErrFile);
 
   // Start the engine.
-  if (!JS_Init()) return 1;
+  const char *message = NULL;
+  if ((message = JS_InitWithFailureDiagnostic())) { fprintf(stderr, "%s\n", message); return 1; }
 
   auto shutdownEngine = MakeScopeExit([]() { JS_ShutDown(); });
 
